### PR TITLE
Importing a private key to create a wallet only supports displaying the address type

### PR DESCRIPTION
**Pull request by shuaiLiWang**
## What does this implement/fix? Explain your changes.
commit1：支持新需求 https://github.com/OneKeyHQ/TaskHub/issues/1287#issuecomment-823813772
commit2：解决通过xpub导入创建钱包时报错问题

## Does this close any currently open issues?  
If it fixes a bug or resolves a feature request, be sure to link to that issue.
https://github.com/OneKeyHQ/TaskHub/issues/1313

## Pull request type
 
_Put an `x` in the boxes that apply_
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## Where has this been tested?
Python and Android
## Any other comments?
…

### CODE REVIEW

File: electrum_gui/android/console.py [hunk 1]
```diff
@@ -3543,17 +3543,21 @@ def recovery_create(self, name, seed, password, bip39_derivation, passphrase="",
...
-        )
+        derivation_path = ""
+        try:
+            derivation_path = self.wallet.get_derivation_path(address)
+            add_type = helpers.get_path_info(derivation_path, PURPOSE_POS)
+        except Exception:
+            script_type = self.wallet.get_txin_type(address)
+            add_type = [str(k) for k, v in PURPOSE_TO_ADDRESS_TYPE.items() if v == script_type]
+            add_type = add_type[0] if add_type else '49'
+            pass
```
Comment (huazhouwang) [inline electrum_gui/android/console.py:3558]: `pass` 多余了

**Review by huazhouwang — COMMENTED** [review]
1. 描述写有两个 commit，可我只看到一个
2. 其实我没能理解这个改动做了什么，达到了什么目的，可以更详细描述下吗？

**Comment by peterQin0826** [comment]
此次commit 是今天提测内容，加急

**Review by huazhouwang — APPROVED** [review]

File: electrum/eth_wallet.py [hunk 2]
```diff
@@ -1042,7 +1042,9 @@ def decrypt_message(self, pubkey: str, message, password) -> bytes:
 
     #The return value is the full path
     def get_derivation_path(self, address):
-        return self.keystore.get_derivation_prefix()
+        if self.keystore and not isinstance(self.keystore, keystore.Imported_KeyStore):
```
Comment (taimanhui) [inline electrum/eth_wallet.py:1045]: 第一个判断应该是`self.keystore is not None`，这样子就是两个not加上and，避免用太多否定的话用两个肯定的判断然后or连接：
```python
if self.keystore is None or is instance(self.keystore, keystone.Imported_Keystore):
    return ""
else:
    return self.keystore.get_derivation_prefix()
```

File: electrum_gui/android/console.py [hunk 3]
```diff
@@ -3543,17 +3543,19 @@ def recovery_create(self, name, seed, password, bip39_derivation, passphrase="",
...
         self._assert_wallet_isvalid()
         derivation_path = self.wallet.get_derivation_path(address)
-        return json.dumps(
-            {
-                "coin": self.wallet.coin,
-                "path": derivation_path,
-                "type": helpers.get_path_info(derivation_path, PURPOSE_POS),
-            }
-        )
+        out_data = {"coin": self.wallet.coin, "path": derivation_path, "type": ""}
```
Comment (taimanhui) [inline electrum_gui/android/console.py:3558]: 非btc的type忘了处理，应该保留原来的做法吧？

File: electrum_gui/android/console.py [hunk 4]
```diff
@@ -2938,7 +2936,16 @@ def create_and_check_wallet(  # noqa
...
-                wallet = Imported_Wallet.from_pubkey_or_addresses(coin, self.config, addresses)
+                if keystore.is_xpub(addresses):
+                    wallet = Standard_Wallet.from_master_key("btc", self.config, addresses)
+                else:
+                    try:
+                        pubkey = ecc.ECPubkey(bfh(addresses)).get_public_key_hex()
+                        addresses = bitcoin.pubkey_to_address(purpose, pubkey)
+                    except Exception:
+                        if not bitcoin.is_address(addresses):
+                            raise util.UnavailableBtcAddr()
```
Comment (taimanhui) [inline electrum_gui/android/console.py:2947]: 这串代码在electrum.wallet.Imported_Wallet里面有的，改对应的`_from_pubkey()`增加对addresses的检查即可。但貌似其实也不用，如果从pubkey里读到的地址都没效的话会报告`No address available`的。

File: electrum_gui/android/console.py [hunk 5]
```diff
@@ -3543,17 +3543,19 @@ def recovery_create(self, name, seed, password, bip39_derivation, passphrase="",
     def get_wallet_derivation_path(self, address: str) -> str:
         """
         Get the current wallet derivation path
+        :param address: receive address as string
```
Comment (taimanhui) [inline electrum_gui/android/console.py:3553]: "receive address"？不是这个意思吧

File: electrum_gui/android/console.py (hunk 3 above)
Comment (shuaiLiWang) [reply electrum_gui/android/console.py:3558]: 非BTC 不需要这个展示字段

File: electrum_gui/android/console.py (hunk 5 above)
Comment (shuaiLiWang) [reply electrum_gui/android/console.py:3553]: 当前钱包的收款地址 我改成receiving address？

Comment (taimanhui) [reply electrum_gui/android/console.py:3553]: 了解，这个应该也是之前说过换地址的功能要求的吧？
补充成「receiving address of the wallet」就好吧。主要是单独一个receive address容易和转账时的目标地址混淆

Comment (shuaiLiWang) [reply electrum_gui/android/console.py]: 好的

File: electrum_gui/android/console.py (hunk 4 above)
Comment (shuaiLiWang) [reply electrum_gui/android/console.py]: ack

File: electrum/eth_wallet.py (hunk 2 above)
Comment (shuaiLiWang) [reply electrum/eth_wallet.py]: ack

File: electrum_gui/android/console.py (hunk 5 above)
Comment (shuaiLiWang) [reply electrum_gui/android/console.py]: 已改

**Review by taimanhui — APPROVED** [review]

**Pull request merged**